src/platform_impl/windows/mod.rs

```diff
@@ -248,3 +253,158 @@ impl Drop for PixelBuffer {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    #![cfg_attr(test, allow(dead_code))]
+    use super::*;
+    use winapi::um::{
+        processthreadsapi::GetCurrentProcess,
+        winnt::HANDLE,
+        winuser::{GetDesktopWindow, GetDC, ReleaseDC, CreateMenu, DestroyMenu},
+    };
+    use winapi::shared::windef::HWND;
+    // Resource consumption tests observe per-process state. To ensure that tests do not interfere
+    // with each other, they must not execute concurrently. The `#[serial]` attribute macro ensures
+    // that tests are run in serial.
+    use serial_test::serial;
+
+    // TODO: Remove the following once `GetGuiResources` is available from the winapi crate.
+    const GR_GDIOBJECTS: u32 = 0x0;
+    const GR_USEROBJECTS: u32 = 0x1;
+    const GR_GDIOBJECTS_PEAK: u32 = 0x2;
+    const GR_USEROBJECTS_PEAK: u32 = 0x4;
+    extern "system" { fn GetGuiResources( hProcess: HANDLE, uiFlags: u32 ) -> u32; }
+
+    struct ObjCount {
+        gdi: u32,
+        user: u32,
```

Comment:
What are `USER` objects, and what is the purpose of testing their count? I ask because the tests will occasionally randomly fail with a mismatched number of `USER` objects between the start and the end. The `GDI` tests seem sound, though.

Here's an example of the `USER` counts failing the tests. 

```
   Finished test [unoptimized + debuginfo] target(s) in 0.07s
     Running S:/programming/cargo_shared_target\debug\deps\winit_blit-e0371a1db547b37e.exe

running 4 tests
test platform_impl::platform_impl::tests::resource_count_unchanged ... ok
test platform_impl::platform_impl::tests::pixelbuffer_blit_resource_leaks ... ok
test platform_impl::platform_impl::tests::resource_count_with_change ... FAILED
test platform_impl::platform_impl::tests::pixelbuffer_new_resource_leaks ... FAILED

failures:

---- platform_impl::platform_impl::tests::resource_count_with_change stdout ----
thread 'platform_impl::platform_impl::tests::resource_count_with_change' panicked at 'assertion failed: obj_count_base.user < obj_count_current.user', src\platform_impl\windows\mod.rs:342:9
stack backtrace:
   0: backtrace::backtrace::trace_unsynchronized
             at C:\Users\VssAdministrator\.cargo\registry\src\github.com-1ecc6299db9ec823\backtrace-0.3.46\src\backtrace\mod.rs:66
   1: std::sys_common::backtrace::_print_fmt
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\sys_common\backtrace.rs:78
   2: std::sys_common::backtrace::_print::{{impl}}::fmt
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\sys_common\backtrace.rs:59
   3: core::fmt::write
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libcore\fmt\mod.rs:1076
   4: std::io::Write::write_fmt<test::helpers::sink::Sink>
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\libstd\io\mod.rs:1537
   5: std::io::impls::{{impl}}::write_fmt<Write>
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\io\impls.rs:176
   6: std::sys_common::backtrace::_print
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\sys_common\backtrace.rs:62
   7: std::sys_common::backtrace::print
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\sys_common\backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\panicking.rs:198
   9: std::panicking::default_hook
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\panicking.rs:215
  10: std::panicking::rust_panic_with_hook
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\panicking.rs:477
  11: std::panicking::begin_panic<str*>
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\libstd\panicking.rs:404
  12: winit_blit::platform_impl::platform_impl::tests::resource_count_with_change::{{closure}}
             at .\src\platform_impl\windows\mod.rs:342
  13: core::ops::function::FnOnce::call_once<closure-0,()>
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\libcore\ops\function.rs:232
  14: serial_test::serial_core
             at C:\Users\Aubrey\scoop\persist\rustup\.cargo\registry\src\github.com-1ecc6299db9ec823\serial_test-0.4.0\src\lib.rs:64
  15: winit_blit::platform_impl::platform_impl::tests::resource_count_with_change
             at .\src\platform_impl\windows\mod.rs:323
  16: winit_blit::platform_impl::platform_impl::tests::resource_count_with_change::{{closure}}
             at .\src\platform_impl\windows\mod.rs:329
  17: core::ops::function::FnOnce::call_once<closure-0,()>
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\libcore\ops\function.rs:232
  18: alloc::boxed::{{impl}}::call_once
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\liballoc\boxed.rs:1076
  19: std::panic::{{impl}}::call_once
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\libstd\panic.rs:318
  20: std::panicking::try::do_call
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\libstd\panicking.rs:297
  21: std::panicking::try
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\libstd\panicking.rs:274
  22: std::panic::catch_unwind
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\libstd\panic.rs:394
  23: test::run_test_in_process
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libtest\lib.rs:541
  24: test::run_test::run_test_inner::{{closure}}
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libtest\lib.rs:450
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- platform_impl::platform_impl::tests::pixelbuffer_new_resource_leaks stdout ----
thread 'platform_impl::platform_impl::tests::pixelbuffer_new_resource_leaks' panicked at 'assertion failed: `(left == right)`
  left: `4`,
 right: `3`: Expected USER object count: 4; observed USER object count: 3', src\platform_impl\windows\mod.rs:379:9
stack backtrace:
   0: backtrace::backtrace::trace_unsynchronized
             at C:\Users\VssAdministrator\.cargo\registry\src\github.com-1ecc6299db9ec823\backtrace-0.3.46\src\backtrace\mod.rs:66
   1: std::sys_common::backtrace::_print_fmt
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\sys_common\backtrace.rs:78
   2: std::sys_common::backtrace::_print::{{impl}}::fmt
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\sys_common\backtrace.rs:59
   3: core::fmt::write
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libcore\fmt\mod.rs:1076
   4: std::io::Write::write_fmt<test::helpers::sink::Sink>
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\libstd\io\mod.rs:1537
   5: std::io::impls::{{impl}}::write_fmt<Write>
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\io\impls.rs:176
   6: std::sys_common::backtrace::_print
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\sys_common\backtrace.rs:62
   7: std::sys_common::backtrace::print
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\sys_common\backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\panicking.rs:198
   9: std::panicking::default_hook
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\panicking.rs:215
  10: std::panicking::rust_panic_with_hook
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\panicking.rs:477
  11: std::panicking::begin_panic_handler
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\panicking.rs:385
  12: std::panicking::begin_panic_fmt
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libstd\panicking.rs:339
  13: winit_blit::platform_impl::platform_impl::tests::pixelbuffer_new_resource_leaks::{{closure}}
             at .\src\platform_impl\windows\mod.rs:379
  14: core::ops::function::FnOnce::call_once<closure-0,()>
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\libcore\ops\function.rs:232
  15: serial_test::serial_core
             at C:\Users\Aubrey\scoop\persist\rustup\.cargo\registry\src\github.com-1ecc6299db9ec823\serial_test-0.4.0\src\lib.rs:64
  16: winit_blit::platform_impl::platform_impl::tests::pixelbuffer_new_resource_leaks
             at .\src\platform_impl\windows\mod.rs:362
  17: winit_blit::platform_impl::platform_impl::tests::pixelbuffer_new_resource_leaks::{{closure}}
             at .\src\platform_impl\windows\mod.rs:367
  18: core::ops::function::FnOnce::call_once<closure-0,()>
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\libcore\ops\function.rs:232
  19: alloc::boxed::{{impl}}::call_once
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\liballoc\boxed.rs:1076
  20: std::panic::{{impl}}::call_once
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\libstd\panic.rs:318
  21: std::panicking::try::do_call
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\libstd\panicking.rs:297
  22: std::panicking::try
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\libstd\panicking.rs:274
  23: std::panic::catch_unwind
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\src\libstd\panic.rs:394
  24: test::run_test_in_process
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libtest\lib.rs:541
  25: test::run_test::run_test_inner::{{closure}}
             at /rustc/feb3536eba10c2e4585d066629598f03d5ddc7c6\/src\libtest\lib.rs:450
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


failures:
    platform_impl::platform_impl::tests::pixelbuffer_new_resource_leaks
    platform_impl::platform_impl::tests::resource_count_with_change

test result: FAILED. 2 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--lib'
```

Reply:
[USER objects](https://docs.microsoft.com/en-us/windows/win32/sysinfo/user-objects) (like [GDI objects](https://docs.microsoft.com/en-us/windows/win32/sysinfo/gdi-objects)) are precious, system-controlled resources. I initially added code to monitor those resources, though I don't recall, why exactly. I may have gotten confused by `GetDC` being a member of `winuser`, even though device contexts are GDI objects.

I have since removed the code to query for and verify USER object counts. The GDI object related code is sufficient.

I've also removed the code that verifies the test code; it appears to work reliably, so there's little benefit in verifying it in the future.